range.c

```diff
@@ -2391,18 +2391,17 @@ empty_region_p(VALUE beg, VALUE end, int excl)
  *    (1..2).overlap?(3..4)      # => false
  *    (1...3).overlap?(3..4)     # => false
  *
- *  This method assumes that there is no minimum value because
- *  Ruby lacks a standard method for determining minimum values.
- *  This assumption is invalid.
- *  For example, there is no value smaller than <tt>-Float::INFINITY</tt>,
- *  making <tt>(...-Float::INFINITY)</tt> an empty set.
- *  Consequently, <tt>(...-Float::INFINITY)</tt> has no elements in common with itself,
- *  yet <tt>(...-Float::INFINITY).overlap?((...-Float::INFINITY))<tt> returns
- *  +true+ due to this assumption.
- *  In general, if <tt>r = (...minimum); r.overlap?(r)</tt> returns +true+,
- *  where +minimum+ is a value that no value is smaller than.
- *  Such values include <tt>-Float::INFINITY</tt>, <tt>[]</tt>, <tt>""</tt>, and
- *  classes without subclasses.
+ *  Note that the method wouldn't make any assumptions about the semi-open
```

Comment:
Rather than introducing a new term "semi-open range," I would prefer "exclusive range," which is used elsewhere.

Reply:
I used it here as a synonym for "beginless or endless" (not for "exclusive range"), yet the statement is related only to beginless one, so I'll use more precise wording.